File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let mergedStr = "", i = 0, j = 0;
    while (i < word1.length || j < word2.length) {
        if (i >= word1.length) return mergedStr + word2.slice(j);
        if (j >= word2.length) return mergedStr + word1.slice(i);

        if (i <= j) {
            mergedStr += word1[i];
            i++;
        } else {
            mergedStr += word2[j];
            j++;
        }
    }
    return mergedStr;
};

File: 1894-merge-strings-alternately/merge-strings-alternately.java

class Solution {
    public String mergeAlternately(String word1, String word2) {
        int i = 0; int j = 0; int n = word1.length();
        int m = word2.length();
        String result = "";

        while (i < n || j < m) {
            if (i < n) {
                result += word1.charAt(i);
                i++;
            }
            if (j < m) {
                result += word2.charAt(j);
                j++;
            }
        }
        return result;
    }
}